core/services/getBranchStatus.js

```diff
@@ -1,26 +1,45 @@
 'use strict';
 
+const logger = require('../../lib/logger');
+
 const { find } = require('lodash');
 const { waterfall } = require('async');
 
 module.exports = (github, POLLING_INTERVAL_MS) => {
   return (repo, branch, cb) => {
     waterfall([
       (next) => github.getBranch(repo, branch, (err, data) => {
-        if (err) return next(err);
+        if (err) {
+          logger.error(`ERROR Getting branch for ${repo} ${branch}`, data)
+          return next(err);
+        }
         next(null, data.object.sha);
       }),
       (sha, next) => github.getProtectedBranchRequiredStatusChecks(repo, branch, (err, data) => {
-        if (err) return next(err);
-        next(null, sha, data.contexts);
+        if (err && err.status == '404') {
```

Comment:
Use strict comparisons 
```suggestion
        if (err && err.status === '404') {
```